source/blender/blenkernel/intern/python_component.c

```diff
@@ -795,6 +815,22 @@ void *BKE_python_component_argument_dict_new(PythonComponent *pc)
         PyList_SetItem(value, i, PyFloat_FromDouble(cprop->vec[i]));
       }
     }
+#define PT_DEF(name, lower, upper) \
+    else if (cprop->type == CPROP_TYPE_ ## upper && cprop->lower) { \
+        ID *id = &cprop->lower->id; \
+        if (id) { \
+            if (!id->py_instance) { \
+                id -> py_instance = pyrna_id_CreatePyObject(id); \
```

Comment:
I'm not entirely certain that it's a safe approach. It seems to work, however, both in standalone and embedded mode.